pkg/config/config.go

```diff
@@ -0,0 +1,176 @@
+	BeeConfigs  map[string]BeeConfig  `yaml:"bee-configs"`
+	Checks      map[string]Check      `yaml:"checks"`
+	Simulations map[string]Simulation `yaml:"simulations"`
+}
+
+// Inherit is struct used for implementing inheritance in Config objects
+type Inherit struct {
+	ParrentName string `yaml:"_inherit"`
+}
+
+// merge combines Config objects using inheritance
+func (c *Config) merge() (err error) {
+	// merge BeeConfigs
+	mergedBC := map[string]BeeConfig{}
+	for name, v := range c.BeeConfigs {
+		if len(v.ParrentName) == 0 {
+			mergedBC[name] = v
+		} else {
+			parent, ok := c.BeeConfigs[v.ParrentName]
+			if !ok {
+				return fmt.Errorf("bee profile %s doesn't exist", v.ParrentName)
+			}
+			p := reflect.ValueOf(&parent).Elem()
+			m := reflect.ValueOf(&v).Elem()
+			for i := 0; i < m.NumField(); i++ {
+				if m.Field(i).IsNil() && !p.Field(i).IsNil() {
+					m.Field(i).Set(p.Field(i))
+				}
+			}
+			mergedBC[name] = m.Interface().(BeeConfig)
+		}
+	}
+	c.BeeConfigs = mergedBC
+
+	// merge NodeGroups
+	mergedNG := map[string]NodeGroup{}
+	for name, v := range c.NodeGroups {
+		if len(v.ParrentName) == 0 {
+			mergedNG[name] = v
+		} else {
+			parent, ok := c.NodeGroups[v.ParrentName]
+			if !ok {
+				return fmt.Errorf("node group profile %s doesn't exist", v.ParrentName)
+			}
+			p := reflect.ValueOf(&parent).Elem()
+			m := reflect.ValueOf(&v).Elem()
+			for i := 0; i < m.NumField(); i++ {
+				if m.Field(i).IsNil() && !p.Field(i).IsNil() {
+					m.Field(i).Set(p.Field(i))
+				}
+			}
+			mergedNG[name] = m.Interface().(NodeGroup)
+		}
+	}
+	c.NodeGroups = mergedNG
+
+	// merge clusters
+	mergedC := map[string]Cluster{}
+	for name, v := range c.Clusters {
+		if len(v.ParrentName) == 0 {
+			mergedC[name] = v
+		} else {
+			parent, ok := c.Clusters[v.ParrentName]
+			if !ok {
+				return fmt.Errorf("bee profile %s doesn't exist", v.ParrentName)
+			}
+			p := reflect.ValueOf(&parent).Elem()
+			m := reflect.ValueOf(&v).Elem()
+			for i := 0; i < m.NumField(); i++ {
+				if m.Field(i).IsNil() && !p.Field(i).IsNil() {
+					m.Field(i).Set(p.Field(i))
+				}
+			}
+			mergedC[name] = m.Interface().(Cluster)
+		}
+	}
+	c.Clusters = mergedC
+
+	return
+}
+
+// ReadDir reads given directory for YAML files and unmarshals them into Config
+func ReadDir(configDir string) (*Config, error) {
+	// read all files from the directory
+	yamlFiles, err := ioutil.ReadDir(configDir)
+	if err != nil {
+		return nil, fmt.Errorf("reading config dir: %w", err)
+	}
+
+	c := Config{
+		Clusters:    make(map[string]Cluster),
+		NodeGroups:  make(map[string]NodeGroup),
+		BeeConfigs:  make(map[string]BeeConfig),
+		Checks:      make(map[string]Check),
+		Simulations: make(map[string]Simulation),
+	}
+
+	for _, file := range yamlFiles {
+		// check if file is YAML
+		fullPath := filepath.Join(configDir + "/" + file.Name())
+		fileExt := filepath.Ext(fullPath)
+		if fileExt != ".yaml" && fileExt != ".yml" {
```

Comment:
this is potentially not needed, since you can just try to unmarshal in `L129` and if that doesn't work return the error. This way configuration can also be provided through environment variables or through `stdin`

Reply:
If it is not yaml it will skip it and continue without an error, which is handy. App does not need to break if there are other files.